portal-impl/src/com/liferay/portal/service/impl/RegionLocalServiceImpl.java

```diff
@@ -68,78 +68,109 @@ public void deleteCountryRegions(long countryId) {
 	}
 
 	@Override
-	public Region deleteRegion(long regionId) {
+	public Region deleteRegion(long regionId) throws PortalException {
 		Region region = regionPersistence.fetchByPrimaryKey(regionId);
 
 		return deleteRegion(region);
 	}
 
 	@Override
-	public Region deleteRegion(Region region) {
+	public Region deleteRegion(Region region) throws PortalException {
+
+		// Region
 
 		regionPersistence.remove(region);
 
-		AddressLocalService.de
-		return null;
+		// Address
+
+		addressLocalService.deleteRegionAddresses(region.getRegionId());
+
+		return region;
```

Comment:
Organizations need to be updated as well. See `CountryLocalServiceImpl.deleteCountry`.